android/src/main/java/com/example/flutter_segment/FlutterSegmentPlugin.java

```diff
@@ -94,34 +86,31 @@ private void setupChannels(FlutterSegmentOptions options) {
 
       // Here we build a middleware that just appends data to the current context
       // using the [deepMerge] strategy.
-      analyticsBuilder.middleware(
-        new Middleware() {
-          @Override
-          public void intercept(Chain chain) {
-            try {
-              if (appendToContextMiddleware == null) {
-                chain.proceed(chain.payload());
-                return;
+      analyticsBuilder.useSourceMiddleware(
+              chain -> {
```

Comment:
Is this supported by the current JDK? I believe this forces to use a Java 8. I believe this Java version should not be supported yet, and it could be part of another PR. The first thing here is to make this Java 7 compatible.